engine/src/main/resources/assets/shaders/newParticle_vert.glsl

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+uniform vec3 position;
+uniform vec3 scale;
+uniform vec4 color;
+
+
+mat4 undoRotation(in mat4 matrix) {
```

Comment:
This is more of a rotation reset rather than undoing it. But where do you use it?